k9mail/src/main/java/com/fsck/k9/crypto/MessageDecryptVerifier.java

```diff
@@ -53,18 +86,28 @@
         return encryptedParts;
     }
 
-    public static List<Part> findSignedParts(Part startPart) {
-        List<Part> signedParts = new ArrayList<Part>();
-        Stack<Part> partsToCheck = new Stack<Part>();
+    public static List<Part> findSignedParts(Part startPart, MessageCryptoAnnotations messageCryptoAnnotations) {
```

Comment:
MessageCryptoAnnotations seems increasingly bad as a name. MessageCryptoParts seems better from it's usage here.